http://sourceware.org/bugzilla/show_bug.cgi?id=9795

When using Task Manager I see that one of the two wmaker processes
(the one started with wmaker --for-real) leaks handles with a rate of 30/sec
right after starting up.

diff --git a/src/xcb_io.c b/src/xcb_io.c
index 913e04e..f58316a 100644
--- a/src/xcb_io.c
+++ b/src/xcb_io.c
@@ -167,6 +167,7 @@ static void process_responses(Display *dpy, int wait_for_first_event, xcb_generi
 				dpy->xcb->pending_requests = req->next;
 				if(!dpy->xcb->pending_requests)
 					dpy->xcb->pending_requests_tail = &dpy->xcb->pending_requests;
+				xcondition_clear(&(req->condition));
 				free(req);
 				reply = error;
 			}
